packages/web/src/components/address-tile/AddressTile.tsx

```diff
@@ -1,49 +1,102 @@
-import { ReactNode, useCallback, useContext } from 'react'
+import { useCallback, useContext } from 'react'
 
-import { shortenSPLAddress } from '@audius/common'
-import { Text, IconCopy, PlainButton } from '@audius/harmony'
+import {
+  BNUSDC,
+  decimalIntegerToHumanReadable,
+  formatUSDCWeiToCeilingCentsNumber,
+  shortenSPLAddress,
+  useUSDCBalance
+} from '@audius/common'
+import {
+  Text,
+  IconCopy,
+  PlainButton,
+  IconComponent,
+  Flex,
+  Box
+} from '@audius/harmony'
+import { BN } from 'bn.js'
 
-import { Icon } from 'components/Icon'
 import { ToastContext } from 'components/toast/ToastContext'
 import { isMobile } from 'utils/clientUtil'
 import { copyToClipboard } from 'utils/clipboardUtil'
 
-import styles from './AddressTile.module.css'
-
 const messages = {
+  usdcBalance: 'USDC Balance',
   copied: 'Copied to Clipboard!'
 }
 
 type AddressTileProps = {
   address: string
-  left?: ReactNode
-  right?: ReactNode
+  iconLeft: IconComponent
+  iconRight?: IconComponent
 }
 
-export const AddressTile = ({ address, left, right }: AddressTileProps) => {
+export const AddressTile = ({
+  address,
+  iconLeft: IconLeft,
+  iconRight: IconRight
+}: AddressTileProps) => {
   const { toast } = useContext(ToastContext)
   const mobile = isMobile()
 
+  const { data: balance } = useUSDCBalance({ isPolling: true })
+  const balanceNumber = formatUSDCWeiToCeilingCentsNumber(
+    (balance ?? new BN(0)) as BNUSDC
```

Comment:
yeah we need to get stuff switched over to FixedDecimal ASAP. This is getting gnarly. :-(

Reply:
Yeah, I chose to do this because we already have this same pattern throughout these flows, but I'll take the AI to move to FixedDecimal here!